internal/devconfig/configfile/env.go

```diff
@@ -25,10 +26,15 @@ func (c *ConfigFile) ParseEnvsFromDotEnv() (map[string]string, error) {
 	if !c.IsdotEnvEnabled() {
 		return nil, fmt.Errorf("env file does not have a .env extension")
 	}
-
-	file, err := os.Open(c.EnvFrom)
+	envFileAbsPath := filepath.Dir(c.AbsRootPath)
+	if filepath.IsAbs(c.EnvFrom) {
+		envFileAbsPath = path.Join(envFileAbsPath, path.Base(c.EnvFrom))
```

Comment:
should this scenario be `envFileAbsPath = c.EnvFrom`?

if the user has set
```
// devbox.json
{
   "env_from": "/my/absolute/path/foo.env`
}
```

Then, I'd think we want to just obey the absolute path directive. 

Perhaps we should error if the env_from path is absolute, because that would mess up any reproducibility of the devbox.json config. But I'll be open to just letting it fly. We can expect our users to use their judgement for when they want to make their config a bit less reproducible. For example, for the sake of some quick changes they may want to experiment with.

Reply:
I agree with the absolute path erroring out case. Reproducibility should be considered here. I'll push a commit to change this case to an error.

Reply:
We allow absolute paths for flakes and plugins. I don't think env should be different. Absolute paths can be important for reproducibility as well.

Reply:
Thinking about it even more, I don't think relative paths are any more reproducible than absolute paths! relative paths are more reproducible only if `.env` file is in the repo, but in that case you are better off just adding to the `env` field in devbox.json

Reply:
@mikeland73 the idea is to prevent unintended mistakes that comes with putting absolute paths. For example, if user A puts `/Users/A/projects/Aproject/.env` and user B has the repo clone in `/Users/B/...` it will break the reproducibility. 
Do you think this is an acceptable risk and we should allow absolute path here?

Reply:
2 reasons:

* erroring on abs path is not backward compatible. This reason alone is a blocker. path of least resistance is always to preserve backward compatibility, especially if it's easier! 
* relative poths also allow you to make it non reproducible! `../path/to/my/.env` would also live outside the repo. Or just `.env` when `.env` is in gititnore.

Why would folks use this feature with `.env` file that is checked in to their repo? (I guess some convenience maybe, but seems low value). The use case for this feature is for people to have _different_ .env files, so by definition reproducibility is not as important.